### PR TITLE
Allow primitive types in disallowed_methods

### LINKED ISSUE #8079: disallowed_method doesn't work for primitive types
### Summary

When trying to add a method on `f32` to `disallowed-methods`, I can't seem to get it to trigger an error. The context here is [this rust-gpu issue](https://github.com/EmbarkStudios/rust-gpu/issues/796).

(as an aside, https://rust-lang.github.io/rust-clippy/master/ says it's spelled "disallowed_methods", in-code it's spelled `clippy::disallowed_method`, and in `clippy.toml` it's spelled `disallowed-methods`, so I'm not sure what to call it).

### Lint Name

disallowed_method

### Reproducer

I tried this code:

clippy.toml:
```toml
disallowed-methods = [
    "core::f32::clamp",
    "std::f32::clamp",
    "f32::clamp",
    "std::time::Instant::now",
]
```

src/main.rs:
```rust
#![warn(clippy::disallowed_method)]
fn main() {
    let _ = 2.0f32.clamp(3.0f32, 4.0f32);
    let _ = std::time::Instant::now();
}
```

I expected to see this happen:

both `Instant::now` and `f32::clamp` to have an error (`Instant::now` only included to show that the warning is indeed working).

Instead, this happened:

Only `Instant::now` has a warning (`warning: use of a disallowed method std::time::Instant::now`), `f32::clamp` isn't detected. I tried three guesses of how to spell `f32::clamp`, all three of them seem to not work.

### Version

```text
rustc 1.56.1 (59eed8a2a 2021-11-01)
binary: rustc
commit-hash: 59eed8a2aac0230a8b53e89d4e99d55912ba6b35
commit-date: 2021-11-01
host: x86_64-unknown-linux-gnu
release: 1.56.1
LLVM version: 13.0.0
```

### PR DESCRIPTION
**Pull request by Alexendoo**
Fixes #8079

changelog: `disallowed_methods`: Now can disallow methods of primitive types

### CODE REVIEW

File: clippy_utils/src/lib.rs [hunk 1]
```diff
@@ -517,18 +518,33 @@ pub fn path_to_res(cx: &LateContext<'_>, path: &[&str]) -> Res {
...
     let tcx = cx.tcx;
     let crates = tcx.crates(());
-    let krate = try_res!(crates.iter().find(|&&num| tcx.crate_name(num).as_str() == krate));
-    let first = try_res!(item_child_by_name(tcx, krate.as_def_id(), first));
+    let first = try_res!(
+        crates
+            .iter()
+            .find(|&&num| tcx.crate_name(num).as_str() == base)
+            .and_then(|krate| item_child_by_name(tcx, krate.as_def_id(), first))
+            .or_else(|| primitive_impl(tcx, base, first))
```
Comment (camsteffen) [inline clippy_utils/src/lib.rs:545]: I think the primitive check should come before iterating the crates, and the crate/primitive cases can collapse into the same code path before calling `item_child_by_name`.

**Review by flip1995 — COMMENTED** [review]
Impl LGTM. Small suggestion for an addition test.

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**